### DISCUSSION TITLE
Article suggestion: Defensive Software

### DISCUSSION THREAD

**Discussion by life00** [PrivSec-dev/privsec.dev#104] [General]
I would like to suggest an article idea. 
The title could be changed if you can think of a better one. 

The topic of the article is about privileged software that is trying to defend against higher privileged software like for example: 
* games with anti-cheats
* banking apps
* DRM
* _maybe something else_

The goal of the article is to explain why such defensive mechanisms against higher privileged software (_on the example of software mentioned above_) may work or may not.

This topic may not be relevant to your project, but I think it's pretty interesting (_at least to me_).
Currently I haven't started working on it and so no sources yet.
In case it is approved to be relevant I may start _slowly_ working on it or someone may start instead of me.
(If it's not approved, its fine)
In the future I am also planning to contribute and write more articles on different topics, if possible.

**Comment by wj25czxj47bu6q** [comment]
Unless I am misunderstanding, this concept sounds nearly if not completely impossible with any current security model.

Also, of the examples you give, the only one that typically runs with any sort of privilege is games with ring 0 (kernel&#8209;level) anti&#8209;cheat.

**Reply by life00** [reply]
Yes, I guess you are right. 
* Anti-cheats use higher level privilege to _defend_
*  banking apps don't actually _defend,_ they obscure and try to detect intrusion
* DRM software by design uses higher privileged stuff to _defend_

In such case maybe there is no point in it. The conclusion is already apparent - any defence mechanisms that are actually in lower privileged software against higher privileged software does not exist or isn't effective.

At least now I know :)

**Comment by life00** [comment]
In such case this short discussion may end.
